clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -1094,8 +1094,19 @@ public void handleSuccessfulResponse(RequestHeader requestHeader, long now, Meta
 
             // Check if any topic's metadata failed to get updated
             Map<String, Errors> errors = response.errors();
-            if (!errors.isEmpty())
+            if (!errors.isEmpty()) {
                 log.warn("Error while fetching metadata with correlation id {} : {}", requestHeader.correlationId(), errors);
+                // Print additional debugging information such as leader id and epoch for partitions fail metadata fetch
+                // with UNKNOWN_TOPIC_OR_PARTITION error
+                if (errors.containsValue(Errors.UNKNOWN_TOPIC_OR_PARTITION)) {
```

Comment:
It seems we can log the source broker for all kinds of errors, and thus there is no need to treat the Unknown_Topic_or_partition specially.
That is, we can simply change the existing log.war above by adding the source broker info.